trucks.js

```diff
@@ -504,8 +504,68 @@ var foodTrucks = [
 ];
 
 // this module should support the following methods:
+
+var _ = require('underscore');
+
 // getTrucks() - return all trucks
+function getTrucks() {
+
+	return foodTrucks;
+}
+
 // getTruck(name) - return the truck object matching 'name'
+function getTruck(name) {
+
+	var match = _.filter(foodTrucks, function(truck){ 
+
+		// return _.contains(truck.name, a);
+		if (truck.name === name) {
+			return truck;
+		}
+	});
+
+	return match;	
+}
+
 // getFoodTypes() - return unique list of all associated food types (underscore has a function to help)
+function getFoodTypes() {
+
+	var types = [];
+
+	var match = _.filter(foodTrucks, function(truck){ 
+
+		for (var i = 0; i < truck.type.length; i++) {
```

Comment:
While _.filter will loop through the entire list, it's meant to take a function that returns true or false based on a truth test. You could do a simple for loop instead of using _.filter here based on how you're getting the initial list here.